app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,13 @@
+class MoviesController < ApplicationController
+  def show
+    @movie = Movie.find(params[:id]) 
+    @actors = @movie.actor_order_age
+
+    if
+      params[:name].present?
```

Comment:
Rather than routing the form back to this action, a more RESTful option that better adheres to SRP would be to route it to a `create` action. In this case we are creating an association between an existing movie and an existing actor. So, we could create a joins controller called MovieActorsController and route the form to a `create` action there. That action would hold this logic - finding the existing movie and actor and then adding the actor to that movie. Then, it can redirect back to the movie show page. 

Without breaking up this controller action, this action is doing too much - it handles too much responsibility (SRP violation)